deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -47,20 +90,42 @@ GrB_Info GxB_MatrixTupleIter_iterate_row
 ) {
 	GB_WHERE1("GxB_MatrixTupleIter_iterate_row (iter, rowIdx)") ;
 	GB_RETURN_IF_NULL(iter) ;
+	if(iter->iterators_size == 0) return GrB_INVALID_VALUE ;
+
+	for(size_t i = 0; i < iter->iterators_size; i++) {
+		_GxB_MatrixTupleIter *iter_cur = &iter->iterators[i] ;
 
-	// Deplete iterator, should caller ignore returned error.
-	_EmptyIterator(iter) ;
+		// Deplete iterator, should caller ignore returned error.
+		_EmptyIterator(iter_cur) ;
 
-	if(rowIdx < 0 || rowIdx >= iter->nrows) {
-		GB_ERROR(GrB_INVALID_INDEX,
-				 "Row index " GBu " out of range; must be < " GBu,
-				 rowIdx, iter->nrows) ;
+		if(rowIdx < 0 || rowIdx >= iter_cur->nrows) {
+			GB_ERROR(GrB_INVALID_INDEX,
+					"Row index " GBu " out of range ; must be < " GBu,
+					rowIdx, iter_cur->nrows) ;
+		}
+
+		GrB_Index rowIdx_sparse ;
+		bool hypersparse_and_row_is_empty = false ;
+		if(iter_cur->sparsity_type == GxB_SPARSE) {
+			rowIdx_sparse = rowIdx ;
+		} else {
+			// GxB_HYPERSPARSE
+			// locate row index is 'Ah'
+			if(!_find_row_index_in_Ah(iter_cur, rowIdx, &rowIdx_sparse)) hypersparse_and_row_is_empty = true ;
+		}
+
+		// incase matrix is hyper-sparse and iterated row is empty, set 'nvals' to 0,
+		// this will cause the next call to 'next' to report depleted
+		iter_cur->nvals = hypersparse_and_row_is_empty ? 0 : iter_cur->A->p[rowIdx_sparse + 1] ;
+		iter_cur->nnz_idx = iter_cur->A->p[rowIdx_sparse] ;
+		iter_cur->row_idx = rowIdx ;
+		if(iter_cur->sparsity_type == GxB_HYPERSPARSE) {
+			iter_cur->sparse_row_idx = rowIdx_sparse ;
+			iter_cur->h_size = iter_cur->A->nvec ;
```

Comment:
Either add a comment explaining this line or introduce a function which will perform this action but will have a name which will explain what's going on here